### ISSUE TITLE
should trigger error for mismatched start / end directives

### ISSUE THREAD

**Issue by asottile** [motet-a/jinjalint#15]
for example

```jinja
{% if True %}
...
{% endfor %}
```

**Comment by asottile** [comment]
oops, my loaner keyboard has a sticky shift key which causes me to shift+enter

the jinja compiler itself produces this error:

```
Encountered unknown tag 'endfor'. Jinja was looking for the following tags: 'elif' or 'else' or 'endif'. The innermost block that needs to be closed is 'if'.
```
